src/main/java/org/opensearch/security/http/OnBehalfOfAuthenticator.java

```diff
@@ -149,63 +182,37 @@ private AuthCredentials extractCredentials0(final RestRequest request) {
         }
 
         final int index;
-        if((index = jwtToken.toLowerCase().indexOf(BEARER_PREFIX)) > -1) { //detect Bearer
+        if(jwtToken != null && (index = jwtToken.toLowerCase().indexOf(BEARER_PREFIX)) > -1) { //detect Bearer
```

Comment:
What is the "-1" doing? You should be able to just use a matcher no?

Reply:
I think the `indexOf` method returns the index of the first occurrence of the specified substring. If the substring is not found, it returns -1.  But yeah, I have changed it into the matcher.